lib/oauthUtil.js

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable complexity, max-statements */
+var http = require('./http');
+var util = require('./util');
+var AuthSdkError = require('./errors/AuthSdkError');
+
+function isToken(obj) {
+  if (obj &&
+      (obj.accessToken || obj.idToken) &&
+      Array.isArray(obj.scopes)) {
+    return true;
+  }
+  return false;
+}
+
+function addListener(eventTarget, name, fn) {
+  if (eventTarget.addEventListener) {
+    eventTarget.addEventListener(name, fn);
+  } else {
+    eventTarget.attachEvent('on' + name, fn);
+  }
+}
+
+function removeListener(eventTarget, name, fn) {
+  if (eventTarget.removeEventListener) {
+    eventTarget.removeEventListener(name, fn);
+  } else {
+    eventTarget.detachEvent('on' + name, fn);
+  }
+}
+
+function loadFrame(src) {
+  var iframe = document.createElement('iframe');
+  iframe.style.display = 'none';
+  iframe.src = src;
+
+  return document.body.appendChild(iframe);
+}
+
+function loadPopup(src, options) {
+  var title = options.popupTitle || 'External Identity Provider User Authentication';
+  var appearance = 'toolbar=no, scrollbars=yes, resizable=yes, ' +
+    'top=100, left=500, width=600, height=600';
+  return window.open(src, title, appearance);
+}
+
+function getWellKnown(sdk) {
+  return http.get(sdk, sdk.options.url + '/.well-known/openid-configuration');
```

Comment:
As discussed, let's write a comment here (and other places) as a reminder that we should swap this out with issuer